src/madwifi.c

```diff
@@ -821,15 +821,16 @@ check_devname (const char *dev)
 
 	if (dev[0] == '.')
 		return 0;
-	
+
 	ssnprintf (buf, sizeof (buf), "/sys/class/net/%s/device/driver", dev);
-	buf[sizeof (buf) - 1] = 0;
+	buf[sizeof (buf) - 1] = '\0';
 
-	memset (buf2, 0, sizeof (buf2));
```

Comment:
You're right of course, in this case Coverity is not smart enough to see that, and neither was I last night ;)

I still like this, but as a cleanup, to prevent the same thing from happening again.